cmd/kubeadm/app/util/config/common.go

```diff
@@ -66,6 +65,24 @@ func ValidateSupportedVersion(gv schema.GroupVersion) error {
 	return nil
 }
 
+// ValidateNotDeprecatedVersion validates that a supported (ValidateSupportedVersion(gv) == nil) GroupVersion is not deprecated.
+// A deprecated API version is such version, that is supported by us, but cannot be used to init/join a cluster. It can only be used
+// for config migration.
+func ValidateNotDeprecatedVersion(gv schema.GroupVersion) error {
```

Comment:
i think we can combine this with `ValidateSupportVersion()`?

A) if `gv` is in the old know versions return "old API" error
B) if `gv` is in the deprecated version list return "deprecated API" error

WDYT?

Reply:
You are right. It's better to combine them. I can even make the resulting function package private, since it's not called outside of the package anyway.